File: examples/swift/test_scope_leak_fix.swift
// Ported from examples/lumen/test_scope_leak_fix.lm by scripts/port_examples.py; edit the Lumen original, not this file.
func count_to_three() -> Int {
    var i = 0
    while i < 3 {
        i = i + 1
    }
    return i
}

print("First call: " + String(count_to_three()))
print("Second call: " + String(count_to_three()))
print("Third call: " + String(count_to_three()))
print("All calls completed successfully!")

File: examples/rust/test_scope_leak_fix.rs
// Ported from examples/lumen/test_scope_leak_fix.lm by scripts/port_examples.py; edit the Lumen original, not this file.
fn count_to_three() -> i64 {
    let mut i = 0;
    while i < 3 {
        i = i + 1;
    }
    return i;
}

fn main() {
    println!("{}", "First call: " + count_to_three().to_string());
    println!("{}", "Second call: " + count_to_three().to_string());
    println!("{}", "Third call: " + count_to_three().to_string());
    println!("All calls completed successfully!");
}
